### PR TITLE
William/typescript

### PR DESCRIPTION
**Pull request by swansontec**
This just has a few fixes, followed by an all-at-once Typescript conversion (which wasn't too bad, actually).

### CODE REVIEW

File: src/commands/login.js [hunk 1]
```diff
@@ -7,14 +7,14 @@ command(
   'account-remove',
   {
     usage: '<username>',
-    help: 'Removes any locally-stored data for the given username',
+    help: 'Deprecated. Use username-remove.',
```
Comment (swansontec) [inline src/commands/login.js]: username-delete

File: src/commands/login.ts [hunk 2]
```diff
@@ -11,7 +11,7 @@ command(
     help: 'Deprecated. Use username-remove.',
     needsContext: true
   },
-  function(console, session, argv) {
+  async function(console, session, argv) {
     if (argv.length !== 1) throw new UsageError(this)
     const username = argv[0]
```
Comment (swansontec) [inline src/commands/login.ts:17]: TODO: Use `await` instead of `return` in all commands.

File: src/commands/pin.ts [hunk 3]
```diff
@@ -41,10 +39,10 @@ command(
     help: 'Creates or changes the PIN for an account',
     needsLogin: true
   },
-  function(console, session, argv) {
+  async function(console, session, argv) {
     if (argv.length !== 1) throw new UsageError(this)
     const pin = argv[0]
 
-    return session.account.changePin(pin)
```
Comment (swansontec) [inline src/commands/pin.ts:48]: Put this in the earlier commit where we fixed deprecated usage?

File: src/commands/recovery2.ts [hunk 4]
```diff
@@ -25,22 +25,20 @@ command(
...
     const username = argv[1]
 
-    const answers = []
+    const answers: string[] = []
     for (let i = 2; i < argv.length; ++i) {
       answers.push(argv[i])
     }
 
     return session.context
-      .loginWithRecovery2(key, username, answers, null, {})
```
Comment (swansontec) [inline src/commands/recovery2.ts:39]: Move to deprecated usage commit?

File: src/commands/wallet.ts [hunk 5]
```diff
@@ -1,12 +1,14 @@
...
+import { command, UsageError } from '../command'
 
 command(
   'wallet-list',
   {
     help: 'Lists the wallets in an account',
     needsAccount: true
   },
-  function(console, session, argv) {
+  function(console, session, argv): void {
```
Comment (swansontec) [inline src/commands/wallet.ts]: Void not necessary here!?

File: src/util/encoding.ts [hunk 6]
```diff
@@ -1,22 +1,20 @@
...
-
 import baseX from 'base-x'
 
 const base58Codec = baseX(
   '123456789ABCDEFGHJKLMNPQRSTUVWXYZabcdefghijkmnopqrstuvwxyz'
 )
 
 export const base58 = {
-  parse(text) {
+  parse(text: string) {
```
Comment (swansontec) [inline src/util/encoding.ts]: No return type!?

**Review by swansontec — COMMENTED** [review]
Changes needed.

**Review by passabilities — APPROVED** [review]
Looks good. Small changes needed.

**Pull request merged**